routingtable/client_interface.go

```diff
@@ -13,10 +13,14 @@ type RouteTableClient interface {
 	ReplacePath(*net.Prefix, *route.Path, *route.Path)
 	UpdateNewClient(RouteTableClient) error
 	Register(RouteTableClient)
+	RegisterWithOptions(RouteTableClient, ClientOptions)
 	Unregister(RouteTableClient)
 	RouteCount() int64
 	ClientCount() uint64
 	Dump() []*route.Route
 	ReplaceFilterChain(filter.Chain)
 	RefreshRoute(*net.Prefix, []*route.Path)
+	Get(pfx *net.Prefix) *route.Route
```

Comment:
Why do we need to increase the surface of this interface? Is the interface used somewhere in the RIS? 
I prefer smaller interfaces. If we want to define a general interface to plugin the AdjRIBIn or AdjRIBOut into the RIS, we should define a smaller interface in the RIS package. In Go we should define the smallest interface as possible in the consumer package.

Reply:
I'm currently implementing a version of the ris that uses both BMP and BGP as a data source. For this I need to have an interface that covers LocRIB and AdjRIBIn, but this is not yet used anywhere in current master. Extending the RouteTableInterface was one possible option to solve my problem.

I talked to @taktv6 off-github about initially and he said he would be fine with this proposed solution, as he wants to refactor the RIB* implementations anyway. If you on the other hand do not feel comfortable with this I can change this PR to only add `RegisterWithOptions()`, `Get()`, `GetLonger()` and `LPM()` to AdjRIBIn (and for the sake of consistency to AdjRIBOut), leafe the RouteTableClient interface alone and create my own little interface with just these four functions I actualy need in the ris package.

Reply:
Since it is no problem in golang to implement multiple interfaces I prefer a new smaller interface in the RIS package. 
> and for the sake of consistency to AdjRIBOut
It may be consistent but does it have any meaning? If AdjRIBOut is not a meaningful RIS table client, why support it?

Reply:
I don't need it ith AdjRIBOut, but it could be used there, as it is a RoutingTable member. For example one could ask the AdjRIBOut what routing would look like for the RIB advertised towards another peer. The "sake of consistency" was ment as "if I'm already adding those methods to AdjRINBIn (which I need for the RIS-case) I could also add them to AdjRIBOut (where I don't need them, but they could be useful to somebody else). Does that make sense? If you'd prefer to keep the PR minimal here as well I can also only explicitly add the function I require for my RIS-use-case.

Reply:
Sorry, had the wrong struct in mind. The AdjRIBOut does make some sense. I thought of the update sender :facepalm:

Reply:
As title, commit message and code is a bit different I have opened a new PR as an alternative option. Does #268 look better? (If you prefer I can also push the changes into this PR)

Reply:
I like the other MR! It keeps the RouteTableClient interface smaller. If there is a need to add more functions to the interface we can do it later.

Reply:
Cool :) Should I push the changes into this PR?